cloudinit/distros/networking.py

```diff
@@ -179,6 +179,26 @@ def wait_for_physdevs(
 class BSDNetworking(Networking):
     """Implementation of networking functionality shared across BSDs."""
 
+    def generate_fallback_config(
+        self, *, blacklist_drivers=None, config_driver: bool = False
+    ):
+        """Generate network cfg v2 for dhcp on the NIC most likely
+        connected."""
```

Comment:
We should document that `config_driver` has no effect on BSD.

Reply:
for now ;)

Reply:
wait, what even is config_driver

Reply:
This is how it's used: https://github.com/canonical/cloud-init/blob/6bd2cda1d3812750027590fe02a5b9415c8f8c9e/cloudinit/net/__init__.py#L439-L442

It adds a `match` clause so that the definition will only match interfaces of that particular driver. IIRC, this is required because Azure's high perf networking presents two interfaces with identical MACs to the instance (one fast one, one slow one, I think) and we need to select only one of them.

The only callsite that passes `config_driver=True` is in the `DataSourceAzure.py` code (which should look familiar, you've changed in this PR :grin:): https://github.com/canonical/cloud-init/blob/6bd2cda1d3812750027590fe02a5b9415c8f8c9e/cloudinit/sources/DataSourceAzure.py#L1505-L1506